parity/db/rocksdb/helpers.rs

```diff
@@ -29,32 +29,32 @@ pub fn compaction_profile(profile: &DatabaseCompactionProfile, db_path: &Path) -
 
 /// Spreads the `total` (in MiB) memory budget across the db columns.
 /// If it's `None`, the default memory budget will be used for each column.
-pub fn memory_per_column(total: Option<usize>) -> HashMap<Option<u32>, usize> {
+pub fn memory_per_column(total: Option<usize>) -> HashMap<u32, usize> {
 	let mut memory_per_column = HashMap::new();
 	if let Some(budget) = total {
 		// spend 90% of the memory budget on the state column, but at least 256 MiB
 		memory_per_column.insert(ethcore_db::COL_STATE, std::cmp::max(budget * 9 / 10, 256));
-		let num_columns = ethcore_db::NUM_COLUMNS.expect("NUM_COLUMNS is Some; qed");
 		// spread the remaining 10% evenly across columns
-		let rest_budget = budget / 10 / (num_columns as usize - 1);
-		for i in 1..num_columns {
+		let rest_budget = budget / 10 / (ethcore_db::NUM_COLUMNS as usize - 1);
+		// todo[dvdplm] relies on `COL_STATE` being the first column?
```

Comment:
Yes, but we would need a migration if we change that and `memory_per_column` would have to be updated in that case. Or do you have some better way in mind?

Reply:
> Or do you have some better way in mind?

I had a vague idea of using a C-style enum for the columns here and see if we could just iterate over it but totally not important.